frontends/p4/symbol_table.cpp

```diff
@@ -176,23 +176,23 @@ void ProgramStructure::pop() {
 void ProgramStructure::declareType(IR::ID id) {
     if (debug) fprintf(debugStream, "ProgramStructure: adding type %s\n", id.name.c_str());
 
-    LOG3("ProgramStructure: adding type " << id);
+    LOG3("ProgramStructure: adding type " << id.toString());
```

Comment:
I had to add a `toString` method here. I do not know how this compiled before. I do not think `ID` had the `<<` operator implemented.

Reply:
We use IDs in many places in LOG calls.
There must be something else going on.